rfcs/0051-stop-transaction-processing.md

```diff
@@ -0,0 +1,158 @@
+include no transaction at all.
+
+## Motivation
+
+In a hard fork scenario, we want to halt the preceding network and produce a new
+genesis ledger for the succeeding network. This new genesis ledger should be
+produced from a stabilised ledger from the preceding network. This is, we define
+a point in time (slot) where the network continues to operate but with no
+"activity". In detail, after this slot, the network continues to produce blocks
+but without including any transactions, and ensuring any fees paid are 0. This
+will run for a certain number of slots, after which the network will stop
+producing blocks. This will allow the network to stabilise and produce a new
+genesis ledger from the last ledger produced by the network.
+
+This feature enables part of this procedure, by allowing users to define a slot
+at which the node will stop accepting any new transactions or including them in
+blocks (in case it is a BP node).
+
+## Detailed design
+
+The procedure to stop processing transactions after a certain slot will act on
+different components of the node, namely, the CLI, the transaction pool, the
+block producer, and the block validator. The procedure will be as follows:
+
+* The CLI will allow a slot at which the node will stop processing transactions
+  to be set. This will be done by setting a new optional flag `--slot-tx-end`
+  that will be used by the transaction pool and the block producer to stop
+  processing transactions after a certain slot. If the flag is not set,
+  the node will continue processing transactions as usual.
+* The node (daemon) will stop accepting new transactions from the client after
+  the slot set by the CLI flag.
+* The transaction pool will stop accepting new transactions after the slot set
+  by the CLI flag.
+* The transaction pool will drop all transactions from the pool after the slot
+  set by the CLI flag.
+* The snark pool will stop accepting new snark works after the slot set by the
+  CLI flag.
+* The snark pool will drop all snark works from the pool after the slot set by
+  the CLI flag.
+* The block producer will stop including transactions in blocks after the slot
+  set by the CLI flag.
+* The block validator will reject blocks that include transactions after the
+  slot set by the CLI flag.
+
+Each of these procedures will be described in detail in the following sections.
+
+### Client submits transaction
+
+When a client sends a transaction to the node daemon, the node will check if the
+current global slot is less than the slot set by the CLI flag `--slot-tx-end`.
+If the transaction is valid and the slot is less than the slot set by the CLI
+flag, the transaction will be added to the transaction pool. If the current
+global slot is equal or greater than the slot set by the CLI flag, the
+transaction will be rejected. The client will be notified of the rejection and
+the reason why the transaction was rejected.
```

Comment:
For better UX it might be advisable to emit a warning some slots prior to the `--slot-tx-end` informnig a user that the doomsday is nigh and their transaction has little chance of making it into the ledger. How many slots prior this warning should appear depends on the throughput, though, so it might be difficult to measure.